pkg/api/api.go

```diff
@@ -51,13 +53,15 @@ func New(profileCatalog Catalog, logger logr.Logger) API {
 // ProfilesHandler is the handler for /profiles requests.
 func (a *API) ProfilesHandler(w http.ResponseWriter, r *http.Request) {
 	query := r.URL.Query().Get("name")
+	var result []profilesv1.ProfileCatalogEntry
 	logger := a.logger.WithValues("endpoint", r.URL.Path, "name", query)
+
 	if query == "" {
-		a.logger.Error(fmt.Errorf("missing query param"), "name param not set", "name", query)
-		a.logAndWriteHeader(w, http.StatusBadRequest)
-		return
+		result = a.profileCatalog.SearchAll()
+	} else {
+		result = a.profileCatalog.Search(query)
```

Comment:
shall we log when searching for query vs all?

Reply:
added logging 👍🏻